src/Orleans/Async/TaskExtensions.cs

```diff
@@ -284,10 +284,18 @@ internal static T WaitForResultWithThrow<T>(this Task<T> task, TimeSpan timeout)
             throw new TimeoutException(String.Format("WithTimeout has timed out after {0}.", timeSpan));
         }
 
-        internal static Task<T> FromException<T>(Exception exception)
+        internal static Task WrapInTask(Action action)
         {
-            var tcs = new TaskCompletionSource<T>(exception);
-            tcs.TrySetException(exception);
+            TaskCompletionSource<bool> tcs = new TaskCompletionSource<bool>();
+            try
+            {
+                action();
+                tcs.TrySetResult(true);
+            }
+            catch (Exception exc)
+            {
+                tcs.TrySetException(exc);
+            }
```

Comment:
Since I cannot merge it (there's conflicts) and you need to rebase, I'll suggest a small optimization for the most frequent code-path:
Do not return a new TCS task for when the action completes without exceptions, and instead return the cached task:

```
internal static Task WrapInTask(Action action)
{
    try
    {
        action();
        return TaskDone.Done;
    }
    catch (Exception exc)
    {
        return FromException<bool>(exc);
    }
}
```

BTW, this means bringing back the `FromException` method that you just deleted because it wasn't used